faucet/src/bin/faucet-shower.rs

```diff
@@ -0,0 +1,220 @@
+            min_polling_delay: Duration::from_millis(opt.min_polling_delay_ms),
+        },
+        &mut loader,
+    )
+    .await?;
+    CapeWallet::new(backend).await
+}
+
+#[async_std::main]
+async fn main() {
+    let opt = Options::from_args();
+    let mut rng = ChaChaRng::from_entropy();
+    let dir = TempDir::new("faucet-shower").unwrap();
+
+    // Create the parent wallet.
+    let parent_dir = [dir.path(), Path::new("parent")].iter().collect();
+    let mut parent = create_wallet(&opt, &mut rng, opt.master_mnemonic.clone(), parent_dir)
+        .await
+        .unwrap();
+
+    // Generate the key which will be used to transfer to the children. Tell it to start a scan
+    // from the default index (the first event) so it can find records created by the faucet event.
+    let parent_key = parent
+        .generate_user_key("parent key".into(), Some(Default::default()))
+        .await
+        .unwrap();
+
+    // While the ledger scan is going, create the child wallets.
+    let children = iter(0..opt.num_wallets)
+        .then(|i| {
+            let mut rng = ChaChaRng::from_rng(&mut rng).unwrap();
+            let dir = &dir;
+            let opt = &opt;
+            async move {
+                let (_, mnemonic) = KeyTree::random(&mut rng);
+                let dir = [dir.path(), Path::new(&format!("child_wallet_{}", i))]
+                    .iter()
+                    .collect();
+                let mut wallet = create_wallet(opt, &mut rng, mnemonic.to_string(), dir)
+                    .await
+                    .unwrap();
+                let key = wallet
+                    .generate_user_key(format!("child key {}", i), None)
+                    .await
+                    .unwrap();
+                (wallet, mnemonic, key)
+            }
+        })
+        .collect::<Vec<_>>()
+        .await;
+
+    // Once we have all the wallets, we need to wait for the ledger scan so that the parent wallet
+    // can discover a record to transfer from.
+    parent.await_key_scan(&parent_key.address()).await.unwrap();
+    let balance = parent.balance(&AssetCode::native()).await;
+    if balance < opt.transfer_size * (opt.num_wallets as u64) {
+        eprintln!(
+            "Insufficient balance for transferring {} units to {} wallets: {}",
+            opt.transfer_size, opt.num_wallets, balance
+        );
+        exit(1);
+    }
+
+    // Print out the generated child mnemonics and keys _before_ we start doing any transfers. If we
+    // panic or get killed for any reason after we have transferred, it is crucial that we have
+    // already reported all of the mnemonics needed to recover the funds.
+    println!(
+        "Transferring {} units each to the following wallets:",
+        opt.transfer_size
+    );
+    for (_, mnemonic, key) in &children {
+        println!("{} {}", mnemonic, key);
+    }
+
+    // Do the transfers.
+    for (_, _, key) in &children {
+        match parent
+            .transfer(
```

Comment:
Can this be a single transfer to all the keys? Using a for-loop might be preferred here since we want to have a clearer error message and avoid the size being too big, but all our `transfer` calls in Cape and Seahorse seem to do single-receiver transfers, so I'm wondering if we actually don't want to do multiple-receiver transfer.

Reply:
This would be really nice, but in CAPE the largest verifying key in the contract has only 3 outputs, so to have more than 3 receiver wallets we would have to break the receivers into chunks of 3 at a time, which I think adds unnecessary complexity.